### ISSUE TITLE
How to exchange access token aftern login with OIDC

### ISSUE THREAD

**Issue by trinhvandat** [matrix-org/synapse#11886]
Hi there, 
I'm trying to get access token of Matrix server with accesstoken that I got from api of OIDC provider (keycloak) by calling api login with type: m.login.token, but I got 403 error.
This is my process detail:
1. I call api get access token of OIDC provider: 
![image](https://user-images.githubusercontent.com/53535112/152093221-326535af-8464-4566-b71e-934f2868feaa.png)
2. I call api /_matrix/client/r0/login with body
![image](https://user-images.githubusercontent.com/53535112/152093308-70058b11-7b4e-4556-b4d4-b1719850abdf.png)
But I got error:
![image](https://user-images.githubusercontent.com/53535112/152093359-f2368426-391c-4eee-8912-adfebfc01cea.png)

How can I resovle it?
Thanks!!!

**Comment by clokep** [comment]
Support questions are generally best asked in #synapse:matrix.org.

The Client-Server specification includes some information about how SSO works in Matrix though: https://spec.matrix.org/latest/client-server-api/#sso-during-user-interactive-authentication

The access token from your SSO provider is not directly used to login via the `m.login.token` type. It needs to be exchanged via an implementation specific flow.

**Comment by Inoir** [comment]
an example of this would be awesome. like gocal said, no response on the pointed channel.
how can you exchange sso token into matrix token?!

**Comment by dominiwe** [comment]
Yeah would be great to get some details here. What I did was to log into a client and take the access token from there. But that's not really a great solution.
